pkg/diagnostics/component_monitoring.go

```diff
@@ -159,6 +192,59 @@ func (c *componentMetrics) PubsubIngressEvent(ctx context.Context, component, pr
 	}
 }
 
+// BulkPubsubIngressEvent records the metrics for a bulk pub/sub ingress event.
+func (c *componentMetrics) BulkPubsubIngressEvent(ctx context.Context, component, topic string, elapsed float64) {
+	if c.enabled {
+		stats.RecordWithTags(
+			ctx,
+			diagUtils.WithTags(appIDKey, c.appID, componentKey, component, namespaceKey, c.namespace, topicKey, topic),
+			c.bulkPubsubIngressCount.M(1))
+
+		if elapsed > 0 {
+			stats.RecordWithTags(
+				ctx,
+				diagUtils.WithTags(appIDKey, c.appID, componentKey, component, namespaceKey, c.namespace, topicKey, topic),
+				c.bulkPubsubIngressLatency.M(elapsed))
+		}
+	}
+}
+
+// BulkPubsubIngressEventEntries records the metrics for entries inside a bulk pub/sub ingress event.
+func (c *componentMetrics) BulkPubsubIngressEventEntries(ctx context.Context, component, topic string, processStatus string, eventCount int64) {
```

Comment:
As per discussion and https://github.com/dapr/dapr/issues/2218#issuecomment-1253428233, intent is to report events inside this bulk call as well. Going one step further, wanted to report as per the status of these events - retry/drop/success. Hence, have an additional func here to report per status.